### PR TITLE
Add gmp/6.1.2

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **gmp/6.1.2**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/gmp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            if self.options.enable_cxx:
+                configure_args.append('--enable-cxx')
+            self._autotools.configure(args=configure_args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        autotools = self._configure_autotools()
+        autotools.make()
+        # INFO: According to the gmp readme file, make check should not be omitted, but it causes timeouts on the CI server.
+        if self.options.run_checks:
```
Comment (SSE4) [inline recipes/gmp/all/conanfile.py]: it doesn't affect `package_id`, right?
if so, remove it in the `package_id` method from the `self.info.options`

Comment (theirix) [reply recipes/gmp/all/conanfile.py]: Exactly, done.

File: recipes/gmp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    _source_subfolder = "source_subfolder"
+    _autotools = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == 'Visual Studio':
+            raise ConanInvalidConfiguration("The gmp package cannot be deployed on Visual Studio.")
```
Comment (danimtb) [inline recipes/gmp/all/conanfile.py]: deployed?

File: recipes/gmp/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            if self.options.enable_cxx:
+                configure_args.append('--enable-cxx')
+            self._autotools.configure(args=configure_args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        autotools = self._configure_autotools()
+        autotools.make()
+        # INFO: According to the gmp readme file, make check should not be omitted, but it causes timeouts on the CI server.
+        if self.options.run_checks:
```
Comment (danimtb) [inline recipes/gmp/all/conanfile.py]: What does this do? Running tests? Is the comment above still applicable? Do we need that option?

Comment (theirix) [reply recipes/gmp/all/conanfile.py]: GMP is a very fragile library and its correctness depends on compiler flags so authors recommended running tests. Conan recipes usually skip tests. The option `run_checks` allows the user to run tests if required.

File: recipes/gmp/all/conanfile.py (hunk 2 above)
Comment (theirix) [reply recipes/gmp/all/conanfile.py]: Reworded with 'built'

File: recipes/gmp/all/conanfile.py (hunk 3 above)
Comment (danimtb) [reply recipes/gmp/all/conanfile.py]: Thanks for the explanation

Comment (danimtb) [reply recipes/gmp/all/conanfile.py]: Thanks for the explanation

**Pull request merged**